### ISSUE TITLE
Unable to validate furigana

### ISSUE THREAD

**Issue by jiru** [Tatoeba/tatoeba2#3145]
**To Reproduce**
1. Go to https://tatoeba.org/en/sentences/show/224073. This particular sentence is Japanese with text "この２ヶ月間、私はとても忙しかった。" and unconfirmed furigana "この２｛｝ヶ月｛かげつ｝間｛かん｝、私｛わたし｝はとても忙｛いそが｝しかった。"
2. Click on Edit sentence.
3. Edit furigana to "この２｛に｝ヶ月｛かげつ｝間｛かん｝、私｛わたし｝はとても忙｛いそが｝しかった。"
4. Furigana is edited, however かげつ spans over 月 which is incorrect.
5. Click Edit again and click the "undo" button to revert furigana
6. Try to edit again furigana to "この２｛に｝ヶ｛か｝月｛げつ｝間｛かん｝、私｛わたし｝はとても忙｛いそが｝しかった。"
7. See error: The provided sentence differs from the original one near “[”.

**Expected behavior**
The first attempt to fix furigana on ２ should produce correct furigana around ヶ月.

**Comment by DJ-Saidez** [comment]
It appears to be resolved from my end. Do you still see it incorrectly?

**Comment by jiru** [comment]
Yes, it is still incorrect. What I see currently is that the furigana is not validated (i.e. it’s the autogenerated version with the warning sign). This is not correct because it lacks furigana above ２ (it should be <ruby>２<rt>に</rt></ruby>). Lacking furigana over a character as simple as ２ is certainly not a big deal, but the point is that it cannot be corrected in a correct way.

**Comment by ckjpn** [comment]
Verified Problem: I tried it to, and it didn't work for me either.

crtmh.GetHubIsGreat,#0

Pasting that sentence into the following, shows the same furigana as tatoeba.org.

**Comment by ckjpn** [comment]
Perhaps the code tatoeba.org is using doesn't allow furigana to be placed above kana.
That would be one thing to check.

**Comment by kumakyoo42** [comment]
The example can be simplified:

- Enter sentence "ヶ" (language japanese)
- Edit sentence, change transcript to "ヶ｛か｝"

**Comment by kumakyoo42** [comment]
> Perhaps the code tatoeba.org is using doesn't allow furigana to be placed above kana. That would be one thing to check.

That's correct. In `sentence-and-translations.dir.js` in function `markupToStored` kana is explicitly excluded from the regex, resulting in furigana to be placed above nothing. Which finally leads to the error message.

A solution might be, to check, if the kanji part is empty and in this case to allow for one kana instead. I'm not sure, if this can be done with a js-regex. Maybe, instead of the regex a function is needed.

**Comment by kumakyoo42** [comment]
Shouldn't this be kept open, because the commits only solve half of the issue?

**Comment by jiru** [comment]
I solved the other half already 🙂

**Comment by kumakyoo42** [comment]
Ah, I thought you only backported the part I did and the "Furigana is edited, however かげつ spans over 月 which is incorrect." part is still open?
